### ISSUE TITLE
Add Start Address functionality for individual strings for Custom Models

### ISSUE THREAD

**Issue by idleup** [xLightsSequencer/xLights#1935]
Ability to define individual start channels for strings for custom models the same way you can for tree and matrix models.

**Comment by computergeek1507** [comment]
You can define individual string lengths, just not individual start channels

**Comment by AzGilrock** [comment]
I haven't looked at the Custom model code lately but Scott what's the point of defining individual string lengths?  I can't figure out what that would be used for.  Don't the numbers on the grid still define the offset from the start channel?  Do we automatically throw the different strings onto new ports for the controller upload?

**Comment by dkulp** [comment]
It's the new ports for the controller upload part.    For models like the giant wreaths and such, splitting it across ports makes a lot of sense.

**Comment by computergeek1507** [comment]
The string count and string lengths are for the controller upload. But I think Matt wants additionally individual start channels for the individual strings so he can assign different controller outputs to separate universes.

**Comment by AzGilrock** [comment]
Yeah so it was what I guessed.  I believe a current limitation for all our models is you can only define one controller port per model and then we increment sequentially so you cannot split across controllers unless you use the manual method which it sounds like what was being requested.  It seems like we are being limited by our parameter grid right now because why couldn't we provide the controller/port feature for anywhere that a start channel can be defined?

**Comment by idleup** [comment]
Just seeing this conversation and Scott is correct, my goal is individual start channels for custom models made with multiple strings for non-sequential addressing. I will update the original request description...

**Comment by AzGilrock** [comment]
Yeah I guess I don't speak well that's what I said.

**Comment by idleup** [comment]
Any status on this? Its not even necessary to get into the ability to define more that one port for the 'Controller Connection' section of the model to resolve this issue, just simply allow the string start addresses to allow address input like '#10.10.17.156:5:222" like it does for Matrix and Tree models.

**Comment by idleup** [comment]
Is there any way this can be implemented?

**Comment by keithsw1111** [comment]
While not impossible there is a lot of complexity in this for a rare use case. You may be able to solve some of this by leaving gaps in the custom model node numbers but without knowing the specifics it is hard to know. You could also break the model down into multiple models ... although that obviously is not the same from a prop manipulation and even rendering perspective.

In summary not saying no but its not on my priority list.

This also will likely cause some issues. Model overlap being one of the most obvious where xlights will spuriously report these models as overlapping any models on the unused channels in between the strings.

**Comment by idleup** [comment]
With a lot of new developers working on the xLight code is anyone interested in tackling this? I do not believe large models controlled by multiple controllers is a rare use case (maybe it was nearly 5 years ago when I first posted this request). I do have work-arounds I have been using but they are Kludgy

**Comment by idleup** [comment]
> Just seeing this conversation and Scott is correct, my goal is individual start channels for custom models made with multiple strings for non-sequential addressing. This is currently possible with Matrix and Tree Models (maybe others too). I ran into issues with this limitation a couple times during the year but I was always too busy to log it as a feature request until today. I will update the original request description...

With a lot of new developers working on the xLight code is anyone interested in tackling this? I do not believe large models controlled by multiple controllers is a rare use case (maybe it was nearly 5 years ago when I first posted this request). I do have work-arounds I have been using but they are Kludgy

**Comment by computergeek1507** [comment]
Unsure how this is complete

**Comment by cybercop23** [comment]
Its not. We closed old items set as Unlikely to be Implemented.
